containerd/utils.go

```diff
@@ -60,3 +60,30 @@ func WithSysctls(sysctls map[string]string) oci.SpecOpts {
 		return nil
 	}
 }
+
+// WithMemoryLimits sets the `Linux.LinuxResources.Memory.Limit` section to the
+// `limit` specified if the `Linux` section is not `nil`. Additionally sets the
+// `Windows.WindowsResources.Memory.Limit` section if the `Windows` section is
+// not `nil`.
+//
+// If hard limit is set on Windows, the soft limit is ignored.
```

Comment:
Can we update the description as well? May be something like:

```
// WithMemoryLimits accepts soft (`memory`) and hard (`memory_max`) limits as parameters and set the desired 
// limits. With `Nomad<1.1.0` releases, soft (`memory`) will act as a hard limit, and if the container process exceeds
// that limit, it will be OOM'ed. With `Nomad>=1.1.0` releases, users can over-provision using `soft` and `hard`
// limits.  The container process will only get OOM'ed if the hard limit is exceeded.
```

Reply:
If you have something better in mind 🙂 , feel free to use that!

Reply:
Updated!